web_infrastructure/jenkins_job.py

```diff
@@ -0,0 +1,332 @@
+    required: true
+  name:
+    description:
+      - Name of the Jenkins job.
+    required: true
+  password:
+    description:
+      - Password to authenticate with the Jenkins server.
+    required: false
+  state:
+    description:
+      - Action to take with the Jenkins job.
+    required: false
+    choices: ['present', 'absent', 'disabled']
+    default: present
+  token:
+    description:
+      - API token used to authenticate alternatively to password.
+    required: false
+  url:
+    description:
+      - Url where the Jenkins server is accessible.
+    required: false
+    default: http://localhost:8080
+  user:
+    description:
+       - User to authenticate with the Jenkins server.
+    required: false
+'''
+
+EXAMPLES = '''
+# Create a jenkins job using basic authentication
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# Create a jenkins job using the token
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    token: asdfasfasfasdfasdfadfasfasdfasdfc
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# Delete a jenkins job using basic authentication
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: absent
+    url: "http://localhost:8080"
+    user: admin
+
+# Delete a jenkins job using the token
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    token: asdfasfasfasdfasdfadfasfasdfasdfc
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# Disable a jenkins job using basic authentication
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: disabled
+    url: "http://localhost:8080"
+    user: admin
+
+# Disable a jenkins job using the token
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    token: asdfasfasfasdfasdfadfasfasdfasdfc
+    state: disabled
+    url: "http://localhost:8080"
+    user: admin
+
+# To re-enable a job just make it be present
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# A production ready example
+- template:
+    src: test-job.xml.j2
+    dest: /tmp/test-job-config.xml
+    mode: 0644
+    owner: jenkins
+    group: jenkins
+
+- jenkins_job:
+    config_file: /tmp/test-job-config.xml
+    name: test-job
+    token: abcdefghijklmnopqrstuvwxyz
+    state: present
+    url: "https://jenkins.mydomain.com"
+    user: my_user
+'''
+
+RETURN = '''
+'''
+
+import os
+
+try:
+    import jenkins
+    python_jenkins_installed = True
+except ImportError:
+    e = get_exception()
+    python_jenkins_installed = False
+
+try:
+    from lxml import etree as ET
+    python_lxml_installed = True
+except ImportError:
+    e = get_exception()
```

Comment:
please remove this line, unused and since the import is later, it causes an error anyway.